iceoryx_hoofs/README.md

```diff
@@ -1,68 +1,76 @@
+|`DeadlineTimer`      |   |   | Polling based timer to check for an elapsed deadline.  |
 |`expected`           |   |   | Our base class used in error handling. Every function which can fail should return an expected. With this the user knows that this function can fail and that they have to do some kind of error handling. We got inspired by the [C++ expected proposal]( http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0323r7.html) and by the [rust error handling concept](https://doc.rust-lang.org/std/result/enum.Result.html). |
-|`filesystem`       |   |   | Implementation of C++17 filesystem features for instance `cxx::perms` to abstract file permissions |
+|`filesystem`         |   |   | Implementation of C++17 filesystem features for instance `cxx::perms` to abstract file permissions |
 |`forward_list`       |   |   | Heap and exception free, relocatable implementation of `std::forward_list` |
+|`function`           |   |   | A stack-based `std::function` replacement based on `storable_function` |
 |`function_ref`       |   |   | C++11 implementation of the next-gen C++ feature `std::function_ref` see [function_ref proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0792r2.html). It behaves like `std::function` but does not own the callable. |
+|`functional_interface` |   |   | Constructs to easily add functional interfaces like `and_then` to object container. |
 |`GenericRAII`        |   |   | This is an abstraction of the C++ RAII idiom. Sometimes you have constructs where you would like to perform a certain task on creation and then again when they are getting out of scope, this is where `GenericRAII` comes in. It is like a `std::lock_guard` or a `std::shared_ptr` but more generic. |
-|`helplets`           |   |   | Implementations of [C++ Core Guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) concepts like `not_null` `Ensures` `Expects` are contained here. Additionally, we are providing some types to verify preconditions at compile time. Think of an int which has to be always greater 5, here we provide types like `greater_or_equal<int, 6>`.|
+|`helplets`           |   |   | Implementations of [C++ Core Guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) concepts like `not_null` are contained here. Additionally, we are providing some types to verify preconditions at compile time. Think of an int which has to be always greater 5, here we provide types like `greater_or_equal<int, 6>`.|
 |`list`               |   |   | Heap and exception free, relocatable implementation of `std::list` |
-|`MethodCallback  `  |   |   | Constructs a MethodCallback from a pointer to a specific object and a pointer to a method of that object, also as `ConstMethodCallback` available |
+|`MethodCallback`     |   |   | Constructs a callback from a pointer to a specific object and a pointer to a method of that object, also as `ConstMethodCallback` available |
```

Comment:
Could you please add a check for maybe obsolete. With the `cxx::function` at our hands and capturing we do not need this anymore.